lib/dolly/mango_index.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+require 'dolly/document'
+
+module Dolly
+  class MangoIndex
+    class << self
+      DESIGN = '_index'
+
+      def all
+        Dolly::Document.connection.get(DESIGN)[:indexes]
+      end
+
+      def create(name, fields, type = 'json')
+        Dolly::Document.connection.post(DESIGN, build_index_structure(name, fields, type))
+      end
+
+      def find_by_fields(fields)
+        all.find do |index_doc|
+          index_doc.dig(:def, :fields).map(&:keys).flatten == fields
+        end
+      end
+
+      def delete_all
```

Comment:
this method is not being used, why did you add it?

Reply:
Added it as completeness to the api, to handle all the possible indexes i was thinking of upon running the create task, delete all the indexes and creating them from scratch, but i'm exploring a way to use the `rev` to update indexes

Reply:
If this is the case, please add a test demonstrating the usage of this.